### ISSUE TITLE
How to integrate wskdeploy into wsk cli ?

### ISSUE THREAD

**Issue by daisy-ycguo** [openwhisk/openwhisk-wskdeploy#92]
We need to think about how to integrate wskdeploy as a part of wsk cli. Shall we integrate them at the source code level ? Or shall we integrate them at build level ?

**Comment by daisy-ycguo** [comment]
Related with Issue #75

**Comment by jthomas** [comment]
I like the `cf` cli approach where you can add plugins to the core CLI to add features. It means you don't have to install lots of different CLI tools for interacting with the system.

**Comment by rabbah** [comment]
+1 for a plugin approach.

**Comment by mrutkows** [comment]
A plug-in approach would be great if we can get the CLI code (framework) into its own project.  The design should be discussed on "dev" and drawn out on our Apache CWIKI.

**Comment by paulcastro** [comment]
@mrutkows agreed.  At the very least we should have an official Go SDK separate from the core repo.

**Comment by mrutkows** [comment]
Making "low" priority until we have a "pluggable" CLI
